lib/onesignal.rb

```diff
@@ -23,6 +23,13 @@ def self.client
     Onesignal::Client.new
   end
 
+  # Exists to allow for realistic testing of initialised configuration
+  # @return [Configuration] The replaced configuration singleton
+  def self.reset_configuration
```

Comment:
I wasn't sure what to do here as the current configuration mechanism doesn't allow for configuration to be reset to initial state.  What do you suggest?